ubelt/util_candidate.py

```diff
@@ -0,0 +1,552 @@
+        prefix, key = path[:-1], path[-1]
+        # *prefix, key = path
+        for k in prefix:
+            d = d[k]
+        d[key] = value
+
+    def __getitem__(self, path):
+        """
+        Get nested value by path
+
+        Args:
+            path (List): list of indexes into the nested structure
+
+        Returns:
+            value
+        """
+        d = self.data
+        prefix, key = path[:-1], path[-1]
+        # *prefix, key = path
+        for k in prefix:
+            d = d[k]
+        return d[key]
+
+    def __delitem__(self, path):
+        """
+        Remove nested value by path
+
+        Note:
+            It can be dangerous to use this while iterating (because we may try
+            to descend into a deleted location) or on leaf items that are
+            list-like (because the indexes of all subsequent items will be
+            modified).
+
+        Args:
+            path (List): list of indexes into the nested structure.
+                The item at the last index will be removed.
+        """
+        d = self.data
+        prefix, key = path[:-1], path[-1]
+        # *prefix, key = path
+        for k in prefix:
+            d = d[k]
+        del d[key]
+
+    def _walk(self, data=None, prefix=[]):
+        """
+        Defines the underlying generator used by IndexableWalker
+
+        Yields:
+            Tuple[List, object] | None:
+                path (List) - a "path" through the nested data structure
+                value (object) - the value indexed by that "path".
+
+            Can also yield None in the case that `send` is called on the
+            generator.
+        """
+        if data is None:  # pragma: nobranch
+            data = self.data
+        stack = [(data, prefix)]
+        while stack:
+            _data, _prefix = stack.pop()
+            # Create an items iterable of depending on the indexable data type
+            if isinstance(_data, self.list_cls):
+                items = enumerate(_data)
+            elif isinstance(_data, self.dict_cls):
+                items = _data.items()
+            else:
+                raise TypeError(type(_data))
+
+            for key, value in items:
+                # Yield the full path to this position and its value
+                path = _prefix + [key]
+                message = yield path, value
+                # If the value at this path is also indexable, then continue
+                # the traversal, unless the False message was explicitly sent
+                # by the caller.
+                if message is False:
+                    # Because the `send` method will return the next value,
+                    # we yield a dummy value so we don't clobber the next
+                    # item in the traversal.
+                    yield None
+                else:
+                    if isinstance(value, self.indexable_cls):
+                        stack.append((value, path))
+
+
+def compatible(config, func, start=0):
```

Comment:
@connorgreenwell I liked your `filter_args` function in your utility file. I'm considering adding a variant of that to ubelt. Pinging you in case you want to take a look / add a review.

Reply:
Seems good to me. Also TIL about the slash in `def foo(x, /, y)`. Interesting feature.